frontend/src/scenes/events/eventDefinitionsLogic.ts

```diff
@@ -0,0 +1,98 @@
+import { kea } from 'kea'
+import api from 'lib/api'
+import { posthogEvents } from 'lib/utils'
+import { EventDefinition, SelectOption } from '~/types'
+import { eventDefinitionsLogicType } from './eventDefinitionsLogicType'
+
+interface EventDefinitionStorage {
+    count: number
+    next: null | string
+    results: EventDefinition[]
+}
+
+interface EventsGroupedInterface {
+    label: string
+    options: SelectOption[]
+}
+
+export const eventDefinitionsLogic = kea<
+    eventDefinitionsLogicType<EventDefinitionStorage, EventDefinition, EventsGroupedInterface, SelectOption>
+>({
+    reducers: {
+        eventStorage: [
+            { results: [], next: null, count: 0 } as EventDefinitionStorage,
+            {
+                loadEventDefinitionsSuccess: (state, { eventStorage }) => {
+                    return {
+                        count: eventStorage.count,
+                        results: [...state.results, ...eventStorage.results],
+                        next: eventStorage.next,
+                    }
```

Comment:
I think we could just return this in the loader? `...values.eventStorage.results`

Reply:
I removed the reducer and just put this in the loader. Feels simpler...